nixos/modules/services/torrent/deluge.nix

```diff
@@ -180,12 +180,13 @@ in {
 
   config = mkIf cfg.enable {
 
-    services.deluge.package = mkDefault (
-      if versionAtLeast config.system.stateVersion "20.09" then
-        pkgs.deluge-2_x
-      else
-        pkgs.deluge-1_x
-    );
+    assertions = [
+      { assertion = versionAtLeast config.system.stateVersion "20.09";
+        message = "stateVersion of 20.09 or greater required for deluge-2_x";
```

Comment:
I think this message might be misleading people into changing there stateVersion. Instead, it should say something like `Deluge 1.x is no longer supported, please set services.deluge.package = pkgs.deluge-2_x;`

Reply:
Thanks.  I see your point about this message being confusing.  I'm not sure the suggested change works though.  The other part of this change makes it so that `deluge-2_x` is the _only_ Deluge available for use.  The trouble (I guess?) is that before 20.09 the NixOS Deluge service ran `deluge-1_x` and 1.x and 2.x aren't compatible on-disk?

Actually, I have no idea.  But I do know that the service used to give you Deluge 1.x if your state version was older than 20.09 and 2.x if your state version was newer.  I think the consequence of my change is that you *cannot run the NixOS Deluge service* anymore if your state version is older than 20.09.

That sounds pretty bad for users on old systems but I'm not about to set up (or write) a tool to upgrade Deluge state from 1.x to 2.x either.

Maybe I can try to capture more of those ideas in this message, at least.

Reply:
Okay so it suddenly seemed silly to try to explain something I didn't actually understand so I went and asked the Deluge IRC channel if it is safe to upgrade from Deluge 1.x to 2.x in-place.  They said yes.  I assumed the previous `stateVersion` switch was intended to avoid the upgrade for old installs because there was incompatible state (after all, that's what `stateVersion` is for right?).  It seems that was an incorrect assumption.  Now I suppose that it was intended to just be less disruptive by leaving older systems being upgraded on the same version of Deluge they had been using and only give the newer version to new installs (though if so this seems not-entirely correct since I could have been running an older system without Deluge, then upgrade, then decide to start using Deluge ...).

So anyway I think what's correct is to just remove the assertion.